### DISCUSSION TITLE
版面宽度定制器

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2628] [插件说明]
描述
---

> 为不同的浏览器尺寸设置不同的 **版面布局**。

地址
---

[https://simpread.ksria.cn/plugins/details/noCdz57Mw6](https://simpread.ksria.cn/plugins/details/noCdz57Mw6)

适用场景
---

1. 多显示器环境（可为不同的显示器设置不同的版面布局）
2. 宽屏环境

引申
---

一个 [典型的使用场景](https://github.com/Kenshin/simpread/issues/774)。

功能
---

1. 监控当前浏览器尺寸，当存在预设时会按照预设的宽度设置当前版面宽度。

2. 可添加多组版面尺寸。

选项
---

![image](https://user-images.githubusercontent.com/81074/127955718-c5cad6d2-26c7-4611-a3c8-918cc372a9d2.png)

![image](https://user-images.githubusercontent.com/81074/127955683-3234d4d1-9196-4b14-8fd7-9c79d9e8bbfb.png)

![image](https://user-images.githubusercontent.com/81074/127955528-a0be63f4-89d3-4b9b-b53c-93b288acd4f1.png)

注意事项
---

使用此方式后，无法通过阅读模式 → 右下角 → 控制栏 来改变宽度。

更新日志
---

### 1.0.1

- `Optimize` 为配合 [插件管理器](https://github.com/Kenshin/simpread/discussions/4221) 的更新方式做了数据结构的优化。

### 1.0.0

1. 监控当前浏览器尺寸，当存在预设时会按照预设的宽度设置当前版面宽度。

2. 可添加多组版面尺寸。

引申
---

> 安装插件或带有插件触发器的插件失效的 [解决方案](https://github.com/Kenshin/simpread/discussions/2342)。

**Comment by Stark-X** [comment]
当宽度一致时，才会触发对应的 规则吗？还是超过设定的宽度就会触发？
高度值貌似没有被使用吧，去掉，感觉更好理解

**Reply by Kenshin** [reply]
> 当宽度一致时

你可以这么理解，当你的浏览器为某个尺寸时，点击**添加** 后，再次出现这个尺寸时就会设置为这个尺寸的分辨率。

![image](https://user-images.githubusercontent.com/81074/129196404-3c39a5d0-01e5-4fb9-b8d2-454a91e7718d.png)

***

主要使用场景：

> 有两个显示器，在竖屏显示器上以 A 布局显示；在主显示器上以 B 布局显示。

**Comment by Stark-X** [comment]
如果我不把窗口全屏，是不是需要把窗口调整为宽度一致。。？

**Reply by Kenshin** [reply]
> 如果我不把窗口全屏，是不是需要把窗口调整为宽度一致。。？

嗨，我感觉你没理解这个插件的使用逻辑，我再重说一次：

把浏览器的尺寸调整到 x * y 后，选择添加，并调整宽度为 10%，这个尺寸就会记录了，当你的浏览器再次调整到 x * y 后，自动会按照 10% 的宽度设置。

**Reply by Stark-X** [reply]
emmm，我没理解错。
已经设置好一个宽度策略，如果没有使用窗口管理工具，当我把一个窗口放到次屏幕，不全屏的情况下，我只有两个选择：
1. 拖动窗口的大小刚好跟之前设置的宽度一致
2. 再添加一个当前的宽度的版面布局策略

把这个匹配规则改成类似 `@media` 这种模式会不会好点？就是当超过某个设定的宽度值时，就应用版面宽度策略，而不是只有宽度值一致时才应用版面宽度策略

**Reply by Kenshin** [reply]
> 把这个匹配规则改成类似 @media 这种模式会不会好点？就是当超过某个设定的宽度值时，就应用版面宽度策略，而不是只有宽度值一致时才应用版面宽度策略

emmm 这个思路不错，我考虑下。

**Comment by Stark-X** [comment]
遇到了问题：点击 添加 -> 保存 -> 刷新，新增的配置不见了

**Reply by Kenshin** [reply]
使用 https://github.com/Kenshin/simpread/discussions/2342 提到的解决方案。
